### ISSUE TITLE
Completion Suggester mappings supported parameters documentation

### ISSUE THREAD

**Issue by andrestc** [elastic/elasticsearch#13584]
In 1.7.1 Im able to create an type with the following mappings:

```json
 {
  "song" : {
        "properties" : {
            "name" : { "type" : "string" },
            "suggest" : { 
                          "type" : "completion",
                          "analyzer" : "standard",
                          "payloads" : true
            }
        }
    }
}
```
But on the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters-completion.html) its stated that it supports only "search_analyzer" and "index_analyzer". I think we should state that "analyzer" is supported or fail when its used.

**Comment by rjernst** [comment]
index_analyzer was deprecated in 1.6 or 1.7 (sorry I don't remember which) and removed in 2.0. The settings are now `analyzer` (which is used for both index and search analyzer), and optionally also `search_analyzer` (which overrides `analyzer` for search time).

**Comment by andrestc** [comment]
Thanks @rjernst. Do you want me to PR this into the documentation?

**Comment by rjernst** [comment]
@andrestc If you see a place `index_analyzer` is still mentioned (on master), then yes please.

**Comment by andrestc** [comment]
@rjernst I was thinking about mentioning the deprecation in 1.7, e.g, [here](https://github.com/elastic/elasticsearch/blob/1.7/docs/reference/search/suggesters/completion-suggest.asciidoc#mapping) like it is done [here](https://github.com/elastic/elasticsearch/commit/41772737ebe77277334fd26f4a46ab6d01f81b8c), if you think its needed.

**Comment by rjernst** [comment]
I thought I had done that, but it looks like I missed it for this setting. It seems unlikely (to me) there would be a 1.7.3 (we just released 1.7.2 and the 2.0 beta is getting closer to a final release).
